packages/pubsub/src/snapshot.js

```diff
@@ -0,0 +1,81 @@
+/*!
+ * Copyright 2014 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*!
+ * @module pubsub/snapshot
+ */
+
+'use strict';
+
+var commonGrpc = require('@google-cloud/common-grpc');
+var util = require('util');
+
+/**
+ *
+ */
+function Snapshot(pubsub, name, options) {
+  if (!name) {
```

Comment:
We usually don't throw in constructors, and instead in the methods that instantiate them. The thought is, `Snapshot` is a private function, so we should assume it is properly invoked, since we are the only ones with access directly to it. The user functions that receive direct user input should do the input-sanity checking.